app/destination/page.js

```diff
@@ -6,37 +6,47 @@ import styles from "@/components/destination/destination.module.css";
 import { AddWishlistItem } from "@/components/destination/AddWishlistItem";
 import PlanetCard from "@/components/PlanetCard";
 import { destinationToGO } from "@/data/destinations";
+import PlanetWishListItem from "@/components/PlanetWishListItem";
 
 // TASK - React 1 week 2 /* moved */
 // Move this to its own file
 
 export const Destinations = () => {
   const [selectedPlanets, onAddPlanet] = useState([]);
-  const [numberOfPlanets, setNumberOfPlanet] = useState(0);
-  const [isPlanetSelected, setIsSelectedPlanet] = useState(false);
-  const onAddOrRemovePlanet = (name, index) => {
+  const [destinationTogo, setDestinationTodo] = useState([]);
+  const onAddOrRemovePlanet = (thumbnail, name) => {
     // TASK - React 1 week 2
     // Implement this function
     // If you press the "ADD PLANET" the selected planet should display "SELECTED"
     // And the counter should update, how many planets are selected (numberOfPlanets)
 
-    onAddPlanet((prev) => {
-      if (prev.includes(name)) {
-        setIsSelectedPlanet(false);
-        setNumberOfPlanet(numberOfPlanets - 1);
-        return prev.filter((item) => item !== name);
-      } else {
-        setIsSelectedPlanet(true);
-        setNumberOfPlanet(numberOfPlanets + 1);
-        return [...prev, name];
-      }
-    });
-
-    console.log(
-      `You seleceted the following planet: ${name}, with the index of ${index}`
+    onAddPlanet((prev) => [
+      ...prev,
+      {
+        thumbnail,
+        name,
+        id: generateRandomId(),
+      },
+    ]);
+  };
+  function addDestination(itemToGo) {
+    setDestinationTodo((prev) =>
+      prev.some((item) => item.id === itemToGo.id)
+        ? prev.filter((item) => item.id !== itemToGo.id)
+        : [...prev, itemToGo]
```

Comment:
This toggle logic for adding/removing an item from the wishlist is clean, expressive, and avoids unnecessary state mutations. Great use of functional updates in setDestinationTodo.